tests/integration-service/gitlab-integration-reporting.go

```diff
@@ -0,0 +1,288 @@
+				testPipelinerun, err = f.AsKubeDeveloper.IntegrationController.WaitForIntegrationPipelineToGetStarted(integrationTestScenarioPass.Name, snapshot.Name, testNamespace)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testPipelinerun.Labels[snapshotAnnotation]).To(ContainSubstring(snapshot.Name))
+				Expect(testPipelinerun.Labels[scenarioAnnotation]).To(ContainSubstring(integrationTestScenarioPass.Name))
+
+				Expect(testPipelinerun.Labels[snapshotAnnotation]).To(ContainSubstring(snapshot.Name))
+
+			})
+		})
+
+		When("Integration PipelineRun is created", func() {
+			It("should eventually complete successfully", func() {
+				Expect(f.AsKubeAdmin.IntegrationController.WaitForIntegrationPipelineToBeFinished(integrationTestScenarioPass, snapshot, testNamespace)).To(Succeed(), fmt.Sprintf("Error when waiting for an integration pipelinerun for snapshot %s/%s to finish", testNamespace, snapshot.GetName()))
+			})
+
+			It("validates the Integration test secnario PipelineRun is reported to merge request CommitStatus, and it pass", func() {
+				timeout = time.Second * 300
+				interval = time.Second * 1
+
+				Eventually(func() string {
+					commitStatuses, _, err := f.AsKubeAdmin.CommonController.Gitlab.GetClient().Commits.GetCommitStatuses(projectID, mrSha, nil)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					for _, commitStatus := range commitStatuses {
+						commitStatusNames := strings.Split(commitStatus.Name, "/")
+						if len(commitStatusNames) > 2 {
+							if strings.Contains(integrationTestScenarioPass.Name, strings.TrimSpace(commitStatusNames[1])) {
+								return commitStatus.Status
+							}
+						}
+					}
+					return ""
+				}, timeout, interval).Should(Equal(integrationPipelineRunCommitStatusSuccess), fmt.Sprintf("timed out when waiting for init PaC MR (branch name '%s') to be created in %s repository", pacBranchName, componentRepoNameForStatusReporting))
+			})
+		})
+
+	})
+})
+
+// createBuildSecret creates a secret of gitlab build
+func createBuildSecret(f *framework.Framework, secretName string, annotations map[string]string, token string) error {
+	buildSecret := v1.Secret{}
+	buildSecret.Name = secretName
+	buildSecret.Labels = map[string]string{
+		"appstudio.redhat.com/credentials": "scm",
+		"appstudio.redhat.com/scm.host":    "gitlab.com",
+	}
+	if annotations != nil {
+		buildSecret.Annotations = annotations
+	}
+	buildSecret.Type = "kubernetes.io/basic-auth"
+	buildSecret.StringData = map[string]string{
+		"password": token,
+	}
+	_, err := f.AsKubeAdmin.CommonController.CreateSecret(f.UserNamespace, &buildSecret)
+	if err != nil {
+		return fmt.Errorf("error creating build secret: %v", err)
+	}
+	return nil
+}
+
+// validateBuildPipelineRunReported within a given project ID and MR ID
+// build PipelineRun name is in MR notes
+func validateBuildPipelineRunReported(f *framework.Framework, pipelineRunName, projectID, mrSha string, mergeResquestID int) {
```

Comment:
This function can't verify merge request note, it check the CommitStatus

Reply:
Maybe we can use
```
// validateReportedAsNoteForPipelineRun within a given project ID and MR ID
// build PipelineRun name is in MR notes
func validateReportedAsMergeRequestNote(f *framework.Framework, projectID, expectedNot string, mergeResquestID int) {

	var timeout, interval time.Duration

	timeout = time.Minute * 10
	interval = time.Second * 2

	Eventually(func() string {
		// Continue here, get as argument MR ID so use in ListMergeRequestNotes
		allNotes, _, err := f.AsKubeAdmin.CommonController.Gitlab.GetClient().Notes.ListMergeRequestNotes(projectID, mergeRequestID, nil)
		Expect(err).ShouldNot(HaveOccurred())
		for _, note := range allNotes {
				if strings.Contains(note, expectedNot) {
					return true
				}
		}
		return false
	}, timeout, interval).Should(BeTrue(), fmt.Sprintf("timed out waiting to validate merge request note (%s) be reported in mergerequest%d notes", expectedNot, mergeRequestID))
}
```
Then it can be used when checking MR note for integration test

Reply:
In fact when get to this part of validation implementation I used the Note object, 
but what is missing for me in notes is the PLR reported `Status`, in the method you suggest above 
it validate if PLR name given `expectedNot` , WDYT?

Reply:
The PLR name is shown twice, such as this [MR](https://gitlab.com/konflux-qe/hacbs-test-project-integration/-/merge_requests/5642), the first one is the PLR is 
```
Starting Pipelinerun test-comp-pac-gitlab-lecgba-on-pull-request-h5mf6 in namespace gitlab-rep-axjb-tenant
You can monitor the execution using the [OpenShift Console](https://console-openshift-console.apps.kalem.clusters.stonesoupengineering.com/k8s/ns/gitlab-rep-axjb-tenant/tekton.dev~v1beta1~PipelineRun/test-comp-pac-gitlab-lecgba-on-pull-request-h5mf6) PipelineRun viewer or through the command line by
using the [opc](https://docs.openshift.com/pipelines/latest/tkn_cli/installing-tkn.html) CLI with the following command:
```
The second is `Pipelines as Code CI/test-comp-pac-gitlab-lecgba-on-pull-request has successfully validated your commit`
So just checking the PLR name can't make sure which one is being verified, so how about to verify note "**Pipelines as Code CI/test-comp-pac-gitlab-lecgba-on-pull-request** has successfully validated your commit"?

Reply:
Resolving this conversation since we've agreed to address this comment in the follow-up PR in order to not hold-up the blocked effort.